File: Golang/WordSearch/WordSearch_test.go
package WordSearch

import "testing"

func TestExist(t *testing.T) {

	if exist([][]byte{{}}, "AB") == false {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	chars := [][]byte{
		{'A', 'B', 'C', 'E'},
		{'S', 'F', 'C', 'S'},
		{'A', 'D', 'E', 'E'}}

	if exist(chars, "ABCCED") {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if exist(chars, "SEE") {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if exist(chars, "ABCB") == false {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/WordSearchTest.py
from unittest import TestCase

from WordSearch import WordSearch


class TestWordSearch(TestCase):
    def test_search(self):
        ws = WordSearch()

        self.assertFalse(ws.exist([[]], "AB"))

        chars = [['A', 'B', 'C', 'E'], ['S', 'F', 'C', 'S'], ['A', 'D', 'E', 'E']]

        self.assertTrue(ws.exist(chars, "ABCCED"))

        self.assertTrue(ws.exist(chars, "SEE"))

        self.assertFalse(ws.exist(chars, "ABCB"))
